pkg/csi_driver/controller.go

```diff
@@ -101,15 +106,66 @@ func (s *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolu
 			return nil, status.Error(codes.AlreadyExists, err.Error())
 		}
 	} else {
+
+		// Get the reserved-IPV4-cidr from the request. If value is not provided, reservedIPV4Range is defaulted to ""
+		reservedIPV4CIDR, ok := req.GetParameters()[paramReservedIPV4CIDR]
+		if ok {
+
+			// Prevents lock acquisition in case the CIDR is invalid
+			validCIDR := s.config.ipAllocator.ValidateCIDR(reservedIPV4CIDR)
+			if !validCIDR {
+				return nil, err
+			}
+
+			reservedIPRange, err := s.reserveIPV4Range(ctx, newFiler, reservedIPV4CIDR)
```

Comment:
instead of `reserve` lets call it:
`preallocate` (?)
`lock` (?)
`earmark` (?)
`designate` (?)
`allot` (?????)
@msau42

Reply:
out of those, i like preallocate the best, but what is the reverse?  unallocate?

Reply:
depreunallocate

Reply:
does hold and release sound appropriate?

Reply:
sgtm